### PR TITLE
feat(storage): add autoclass samples

### LINKED ISSUE #2707: storage: add samples for Autoclass
Add samples demonstrating getting and setting Autoclass feature for a bucket. Currently, Autoclass supports  **new buckets** only - a bucket can be enabled for Autoclass only at creation, once disabled it cannot be enabled again
- region tags: `storage_get_autoclass` and `storage_set_autoclass`

canonical samples (internal): [go/gcs-autoclass-canonical-sample](http://go/gcs-autoclass-canonical-sample)
details and coordination (internal): [go/gcs-dpe-autoclass](http://go/gcs-dpe-autoclass)
python-storage example: https://github.com/googleapis/python-storage/pull/791

Please wait for approval from myself before merging/releasing so we can coordinate around release dates.

### PR DESCRIPTION
**Pull request by cojenco**
Fixes #2707

Requires https://github.com/googleapis/google-cloud-go/pull/6828
Tests pass locally with library feature support

### CODE REVIEW

**Review by BrennaEpp — COMMENTED** [review]
Thanks Cathy :)

File: storage/buckets/set_autoclass.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+	"cloud.google.com/go/storage"
+)
+
+// setAutoclass sets the Autoclass configuration for a bucket.
+// See https://cloud.google.com/storage/docs/using-autoclass for more information.
+
+// Note: Only update requests that disable Autoclass are currently supported.
+// To enable Autoclass, you must set it at bucket creation time.
+func setAutoclass(w io.Writer, bucketName string, toggle bool) error {
```
[the remark this answers is not in the document]
Comment (cojenco) [reply storage/buckets/set_autoclass.go:32]: Great call out. I had a discussion with nstock@ while planning the canonical samples. As we're aware that there's been individual enable/disable samples in the past, the TW team is working towards integrating to a single setXXX doc section that hosts a single setXXX sample. The pros to this change includes a smoother flow in the docs, less redundancy, samples maintainability, etc. 

I think it's best to keep the parameter. I'll go ahead and rename it to `value` as you suggested. We've also considered in the samples planning V2 Autoclass, in which both enabling/disabling is supported for existing buckets. With the parameter, we'll be able to showcase and test both enabling and disabling.

File: storage/buckets/get_autoclass.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	defer client.Close()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+
+	attrs, err := client.Bucket(bucketName).Attrs(ctx)
+	if err != nil {
+		return fmt.Errorf("Bucket(%q).Attrs: %v", bucketName, err)
+	}
+	fmt.Fprintf(w, "Autoclass enabled is set to %v for %v at %v", attrs.Autoclass.Enabled, bucketName, attrs.Autoclass.ToggleTime)
```
[the remark this answers is not in the document]
Comment (cojenco) [reply storage/buckets/get_autoclass.go]: Updated both, yep this sounds much better, thanks ;)

File: storage/buckets/get_autoclass.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/storage"
+)
+
+// getAutoclass gets the Autoclass configuration for a bucket.
+// See https://cloud.google.com/storage/docs/using-autoclass for more information.
```
[the remark this answers is not in the document]
Comment (cojenco) [reply storage/buckets/get_autoclass.go:28]: good question, hmmmm I don't feel strongly but am inclined to include the link. The sample will be published on that page, but it'll also be published under https://cloud.google.com/storage/docs/samples and potentially discovered through GitHub? Let me know your thoughts

File: storage/buckets/set_autoclass.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (BrennaEpp) [reply storage/buckets/set_autoclass.go]: Awesome, thank you for the in-depth response. I agree that integrating them is best.

**Pull request merged**